### PR TITLE
docs: split rock explanation into its own document

### PR DESCRIPTION
**Pull request by dboddie**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----
See #275.

### CODE REVIEW

File: docs/explanation/rocks.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+For example, Docker can be used to run a ROCK with the usual command line
+syntax:
+
+.. code:: bash
+
+   docker run <rock> ...
+
+Interoperability between ROCKs and other containers also extends to the way
+that container images are built. This enables the use of existing build
+recipes, such as Dockerfiles, with ROCKs. It also opens up the possibility of
```
Comment (cjdcordeiro) [inline docs/explanation/rocks.rst:22]: this 2nd sentence

> This enables the use of existing build ￼recipes, such as Dockerfiles, with ROCKs.

is misleading. You can't use Dockerfile with Rockcraft. I see the confusion as a re-write of the old paragraph, which wasn't super clear either. But the point is: even if you have your own Dockerfile and want to keep building Docker images, you can use ROCKs as bases within your Dockerfile, cause they are as valid as any other Docker base image

File: docs/index.rst [hunk 2]
```diff
@@ -1,45 +1,22 @@
...
-   </h4>
-
-.. |br| raw:: html
-
-   <br />
-
 Welcome to Rockcraft's documentation!
 ===========================================
 
+Rockcraft is a tool to create :ref:`rocks_explanation`, a new generation
```
Comment (cjdcordeiro) [inline docs/index.rst:4]: have the guidelines changed? We've used the previous format on request. And TBH, the use of bold text makes it more readable and structured IMO.

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
thanks. I just have a small comment.

besides that, it would be nice if we could generate RTD previews for every PR. I've seen this in other repos - example: https://github.com/canonical/ubuntu-cloud-docs/pull/38#issuecomment-1531266342

File: docs/explanation/rocks.rst (hunk 1 above)
Comment (dboddie) [reply docs/explanation/rocks.rst:22]: I'll reword it to clarify.

File: docs/index.rst (hunk 2 above)
Comment (dboddie) [reply docs/index.rst:4]: I'll put the bold back in.

**Comment by dboddie** [comment]
> besides that, it would be nice if we could generate RTD previews for every PR.

See https://canonical-rockcraft--283.com.readthedocs.build/en/283/ for this PR. I've enabled it for all pull requests.

File: docs/explanation/rocks.rst (hunk 1 above)
Comment (cjdcordeiro) [reply docs/explanation/rocks.rst]: ty

**Review by cjdcordeiro — APPROVED** [review]
Great, thanks

File: docs/explanation/rockcraft.rst [hunk 3]
```diff
@@ -11,22 +12,6 @@ behind Rockcraft.
 
 
 
-What is a ROCK?
```
[the remark this answers is not in the document]
Comment (dboddie) [reply docs/explanation/rockcraft.rst:14]: I renamed the title in 9f8b609.

**Review by tigarmo — APPROVED** [review]
